### DISCUSSION TITLE
Mod List never saves properly.

### DISCUSSION THREAD

**Discussion by JayXsane** [tsunamods-codes/7th-Heaven#182] [Q&A] [answered]
This is an odd one, and I'm not sure really what could fix this because my files look identical but;
Whenever there's _any update_ to my mod list, whether it's because I updated a single mod, or close the program and reopen it, or reboot my computer, etc; it causes my modlist to shift around and randomize. I've added screenshots of what the modlist changes to, and what the modlist should look like when I reorganize it. This is on saved profiles none-the-less. So I have saved profiles with the correct list, but when I close and reopen the program, it shifts everything around on that list and when I try to check my profile's internal file, everything looks fine.

** Versions **
[X] Tested on latest stable version

**To Reproduce**
Happens any time I update a mod, or close/reopen the program (including and excluding rebooting my PC). This happens in and outside Admin Mode.

**Expected behavior**
When saving the Profile, I expect the list to remain intact regardless if I update a mod or relaunch 7th Heaven. Certain mods need to be in certain spots, which can cause bugs/undesired outcomes.

**Screenshots**

Whenever I update a mod and/or Relaunch the Program:
![7th Heaven v3 3 1 0 - Mod Manager for Final Fantasy 7 ( Maintained by the Tsunamods Team )  2   2024-02-15 1_24_36 PM](https://github.com/tsunamods-codes/7th-Heaven/assets/18487559/6e387df0-6448-4ef8-b983-99cefae2f51d)

The Saved Profile:
![7th Heaven v3 3 1 0 - Mod Manager for Final Fantasy 7 ( Maintained by the Tsunamods Team )  1   2024-02-15 1_28_55 PM](https://github.com/tsunamods-codes/7th-Heaven/assets/18487559/fe29b23b-4082-4aff-b535-bc006aef5304)

**Desktop (please complete the following information):**
 - OS: Windows 11 Home
 - Version: 23H2
 - Build: 22635.3140

**Additional context**
A workaround I've found, albeit not perfect, is to save 2 Profiles with the same list. When the program opens, flip to the profile with the correct list, delete the profile you first loaded, then re-save the profile again so it has 2 profiles saved.
This only works for when reopening the program however. If you update a mod, the Profiles don't retain the update on either saved profile, so you'd still need to update the mod(s), then reorganize the lists manually, then save 2 profiles again. Also keep in mind that if you're actively on (example) Profile 1, and you attempt to Delete Profile 1, it won't let you. You need to first move to Profile 2, Delete Profile 1 then re-save Profile 1 with the corrected list.

**Comment by julianxhokaxhiu** [comment, marked as answer]
This is not a bug, it's a feature. It's called Auto Sorting and it's an option that is enabled by default on the General Settings window. The idea is that auto-sorting removes from you all possible bugs you might encounter which otherwise you could come to us for support and we would suggest you to use the Auto-Sorting anyway. Why that? Because the order matters to load them properly and mods know how to detect each other and put themselves in the right place. I highly suggest you to make use of it instead of playing around, disabling it and going manual means you'll be on your own. Hope this helps, cheers and enjoy!
